go/saltpack/decrypt.go

```diff
@@ -160,13 +164,22 @@ func (ds *decryptStream) tryVisibleReceivers(hdr *EncryptionHeader, ephemeralKey
 func (ds *decryptStream) tryHiddenReceivers(hdr *EncryptionHeader, ephemeralKey BoxPublicKey) (BoxSecretKey, BoxPrecomputedSharedKey, []byte, int) {
 	secretKeys := ds.ring.GetAllSecretKeys()
 
+	for _, r := range hdr.Receivers {
+		if len(r.ReceiverKID) == 0 {
+			ds.mki.NumAnonReceivers++
+		}
+	}
+
 	for _, secretKey := range secretKeys {
+
 		shared := secretKey.Precompute(ephemeralKey)
 
 		for i, r := range hdr.Receivers {
-			keysRaw, err := shared.Unbox(ds.nonce.ForKeyBox(), r.Keys)
-			if err == nil {
-				return secretKey, shared, keysRaw, i
+			if len(r.ReceiverKID) == 0 {
```

Comment:
This `if` statement means we don't try to decrypt named receivers as though they might also be anonymous receivers, do I have that right? Is there any reason we might want to allow that? Weirdo tricks someone might want to pull with making a receiver _appear_ public but actually be anonymous?